### PR TITLE
Add validation, sidebar, login. Improve design.

### PR DESCRIPTION
**Pull request by VadimAlkhimenok**

### CODE REVIEW

File: src/components/auth/Auth.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    if (!isCorrectEmail(this.state.email)) {
+      this.setState({
+        errorEmail: 'Incorrect email',
+        isErrorEmail: true,
+      });
+    }
+
+    if (isCorrectPassword(this.state.email, this.state.password)) {
```
Comment (maximsan) [inline src/components/auth/Auth.js:56]: I believe the condition should be apposite if the password is correct

File: src/components/auth/Auth.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        <ButtonsGroup>
+          <Button name='Confirm' color={color.green} handleClick={this.handleClickAuth} />
+        </ButtonsGroup>
+      </form>
+    </div>
+  );
+
+  render() {
+    return this.auth();
```
Comment (maximsan) [inline src/components/auth/Auth.js:95]: don't see a  real need to create a method for this

File: src/components/modals/modal-register/ModalRegister.js [hunk 3]
```diff
@@ -10,28 +10,81 @@ import { ModalTemplate } from '../modal-template/ModalTemplate';
...
+    skype: '',
+    startDate: '',
+    age: '',
+    errorEmptyInput: '',
+    errorImportant: false,
+  };
+
+  handleClickSubmit = (event) => {
+    event.preventDefault();
+    console.log('true', isEmptyInput(this.state));
```
Comment (maximsan) [inline src/components/modals/modal-register/ModalRegister.js:35]: remove

File: src/components/modals/modal-register/ModalRegister.js [hunk 4]
```diff
@@ -10,28 +10,81 @@ import { ModalTemplate } from '../modal-template/ModalTemplate';
...
+  render() {
+    return (
+      <ModalTemplate handleOpenCloseModal={this.props.handleClickRegister}>
+        <div className={classes.ModalRegister}>
+          <Title text={this.props.title} />
+
+          <InputsGroup>
+            {inputsModalData.map((input, idx) => (
+              <Input
+                key={idx}
```
Comment (maximsan) [inline src/components/modals/modal-register/ModalRegister.js:62]: don't use method index as a key

File: src/components/modals/modal-template/ModalTemplate.module.css [hunk 5]
```diff
@@ -12,17 +12,17 @@
...
 
 .Wrapper {
   display: flex;
   justify-content: center;
   position: absolute;
-  background-color: rgb(145, 145, 145);
+  background-color: rgb(201, 201, 201);
   z-index: 10;
-  height: 100vh;
+  height: 109%;
```
Comment (maximsan) [inline src/components/modals/modal-template/ModalTemplate.module.css:25]: looks weird

File: src/components/select/Select.js [hunk 6]
```diff
@@ -3,23 +3,27 @@ import PropTypes from 'prop-types';
...
       <label htmlFor={title} className={classes.Label}>
         {title}
       </label>
 
-      <select className={classes.Select} id='fDirection' disabled={disabled}>
-        {options.map((option, idx) => (
-          <option key={idx}>{option}</option>
+      <select className={classes.Select} id='fDirection' disabled={disabled} onChange={handleChange} name={title}>
+        {options.map((option, index) => (
+          <option key={`${index}select`}>{option}</option>
```
Comment (maximsan) [inline src/components/select/Select.js:15]: I will propose you to use this `'${option}${index}'`  as a key

File: src/components/side-bar/side-bar-main/SidebarMain.js [hunk 7]
```diff
@@ -1,6 +1,48 @@
...
+  return open ? (
+    <>
+      <div className={classes.Wrapper} onClick={handleClick}></div>
+      <div className={classes.SidebarMain}>
+        <ul className={classes.SidebarList}>
+          <li>
+            <a href='/' onClick={handleClick}>
+              Link
+            </a>
+          </li>
```
Comment (maximsan) [inline src/components/side-bar/side-bar-main/SidebarMain.js:15]: why do you not use `Link` from 'react-router-dom' ?

File: src/pages/member-progress/MemberProgress.js [hunk 8]
```diff
@@ -69,9 +67,9 @@ export default class MemberProgress extends Component {
         ))}
       </HeaderLine>
 
-      {this.state.tasksProgress.map((progress, idx) => (
-        <Lines key={idx}>
-          <Line width={10} text={idx + 1} />
+      {this.state.tasksProgress.map((progress, index) => (
+        <Lines key={`${progress.userTaskId}progress`}>
```
Comment (maximsan) [inline src/pages/member-progress/MemberProgress.js:71]: `userTaskId` is enough I believe

File: src/pages/members-tasks/MemberTasks.js [hunk 9]
```diff
@@ -58,10 +57,10 @@ export default class MemberTasks extends Component {
         ))}
       </HeaderLine>
 
-      {this.state.memberTasks.map((task, idx) => {
+      {this.state.memberTasks.map((task, index) => {
         return (
-          <Lines key={idx}>
-            <Line width={5} text={idx + 1} />
+          <Lines key={`${task.taskId}task`}>
```
Comment (maximsan) [inline src/pages/members-tasks/MemberTasks.js:62]: taskId is enough

File: src/pages/members/Members.js [hunk 10]
```diff
@@ -96,15 +87,15 @@ export default class Members extends Component {
...
         ))}
       </HeaderLine>
 
-      {this.state.members.map((member, idx) => {
+      {this.state.members.map((member, index) => {
         return (
-          <Lines key={idx}>
-            <Line width={3} text={idx + 1} />
+          <Lines key={`${member.userId}member`}>
+            <Line width={3} text={index + 1} />
```
Comment (maximsan) [inline src/pages/members/Members.js:98]: userId is enough

File: src/components/auth/validation/ValidationAuth.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+export const isCorrectEmail = (email) => {
+  const regExp = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return regExp.test(email);
+};
+
+export const isCorrectPassword = (email, password) => {
+  if (email !== '' && password === '') {
+    return true;
+  }
+};
```
Comment (maximsan) [inline src/components/auth/validation/ValidationAuth.js:12]: add unit tests

File: src/components/auth/validation/ValidationAuth.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+export const isEmptyInput = (email, password) => (email === '' && password === '' ? true : false);
+
+export const isCorrectEmail = (email) => {
+  const regExp = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return regExp.test(email);
+};
```
Comment (maximsan) [inline src/components/auth/validation/ValidationAuth.js:6]: add unit tests

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello! Left some notes. Please fix them! Thanks!

File: src/components/auth/Auth.js (hunk 1 above)
Comment (VadimAlkhimenok) [reply src/components/auth/Auth.js:56]: Done.

File: src/components/auth/Auth.js (hunk 2 above)
Comment (VadimAlkhimenok) [reply src/components/auth/Auth.js:95]: Done.

File: src/components/modals/modal-register/ModalRegister.js (hunk 3 above)
Comment (VadimAlkhimenok) [reply src/components/modals/modal-register/ModalRegister.js:35]: Done.

File: src/pages/members/Members.js (hunk 10 above)
Comment (VadimAlkhimenok) [reply src/pages/members/Members.js:98]: Done.

File: src/pages/members-tasks/MemberTasks.js (hunk 9 above)
Comment (VadimAlkhimenok) [reply src/pages/members-tasks/MemberTasks.js:62]: Done.

File: src/pages/member-progress/MemberProgress.js (hunk 8 above)
Comment (VadimAlkhimenok) [reply src/pages/member-progress/MemberProgress.js:71]: Done.

File: src/components/side-bar/side-bar-main/SidebarMain.js (hunk 7 above)
Comment (VadimAlkhimenok) [reply src/components/side-bar/side-bar-main/SidebarMain.js:15]: Done.

File: src/components/select/Select.js (hunk 6 above)
Comment (VadimAlkhimenok) [reply src/components/select/Select.js:15]: Done.

File: src/components/modals/modal-template/ModalTemplate.module.css (hunk 5 above)
Comment (VadimAlkhimenok) [reply src/components/modals/modal-template/ModalTemplate.module.css:25]: Done.

File: src/components/modals/modal-register/ModalRegister.js (hunk 4 above)
Comment (VadimAlkhimenok) [reply src/components/modals/modal-register/ModalRegister.js:62]: Done.

File: .idea/DIMS-UI-10.iml [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.tmp" />
+      <excludeFolder url="file://$MODULE_DIR$/temp" />
+      <excludeFolder url="file://$MODULE_DIR$/tmp" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (maximsan) [inline .idea/DIMS-UI-10.iml:12]: add to gitignore, delete from repo

File: .idea/codeStyles/Project.xml [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+      </indentOptions>
+    </codeStyleSettings>
+    <codeStyleSettings language="Vue">
+      <option name="SOFT_MARGINS" value="120" />
+      <indentOptions>
+        <option name="CONTINUATION_INDENT_SIZE" value="2" />
+      </indentOptions>
+    </codeStyleSettings>
+  </code_scheme>
+</component>
```
Comment (maximsan) [inline .idea/codeStyles/Project.xml:59]: add to gitignore, delete from repo

File: .idea/codeStyles/codeStyleConfig.xml [hunk 15]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <state>
+    <option name="USE_PER_PROJECT_SETTINGS" value="true" />
+  </state>
+</component>
```
Comment (maximsan) [inline .idea/codeStyles/codeStyleConfig.xml:5]: add to gitignore, delete from repo

File: .idea/misc.xml [hunk 16]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="JavaScriptSettings">
+    <option name="languageLevel" value="JSX" />
+  </component>
+</project>
```
Comment (maximsan) [inline .idea/misc.xml:6]: add to gitignore, delete from repo

File: .idea/modules.xml [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/DIMS-UI-10.iml" filepath="$PROJECT_DIR$/.idea/DIMS-UI-10.iml" />
+    </modules>
+  </component>
+</project>
```
Comment (maximsan) [inline .idea/modules.xml:8]: add to gitignore, delete from repo

File: .idea/vcs.xml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (maximsan) [inline .idea/vcs.xml:6]: add to gitignore, delete from repo

**Review by maximsan — APPROVED** [review]
Hello, please don't forget to remove extra files, add them to ignore. I will approve and merge it.

**Pull request merged**